### ISSUE TITLE
why is no certificate needed in node configuration?

### ISSUE THREAD

**Issue by AndySchroder** [ZeusLN/zeus#253]
In the app when I try to setup a node, it asks for a host, port, and macaroon. No certificate is asked for like when I use the grpc interface in python.

Why do you not check the certificate? Seems insecure.

**Comment by kaloudis** [comment]
It's bad practice to connect to your node directly over the plain internet with an open port on your router, so verification should happen when you connect to your private VPN or Tor hidden service.

Your concern is valid though, and could prove to be a problem if you're connecting over your local network when you have a spook on it. we're looking at adding certification verification as an extra level of security as we implement Tor directly into the app. It'll take some time because it has been done with native module code.

**Comment by AndySchroder** [comment]
Is there some issue with the encryption that LND uses that make it weaker than a VPN's encryption?

I realize there is a rough identity privacy loss(the IP only of your LND machine) from whatever service provider you use to access the internet from your phone at the time, but how does that differ from loosing rough identity privacy by accessing your own self hosted e-mail server at the same time?

Seems like you would still want to validate a certificate if you connect over tor, but not to a .onion hidden site.

Bottom line, I think this should have a multi layer privacy, as well as a choose your own privacy model. Tor is not totally decentralized and has its own weaknesses. All layers should be as secure as possible so that when one a weakness occurs in one layer, everything isn't exposed.

**Comment by kaloudis** [comment]
It's just an avenue for being DOS'd or having funds compromised.

I agree with you about layers. I'm working on adding cert validation as I add in built-in Tor support to both the iOS and Android app.

Thanks for your concern, it's well-founded

**Comment by kaloudis** [comment]
Brief update. Tried using `rn-nodeify` with `https` for a custom agent in `axios` to no avail. Going to take a look at this but I'm not 100% convinced it will work: https://code.janeasystems.com/nodejs-mobile

There is also talks to add RPC, so maybe we can alleviate this concern there.

**Comment by kaloudis** [comment]
Found a good option with `rn-fetch-blob`
PR is progress here: https://github.com/ZeusLN/zeus/pull/270

Users will have to manually add the SSL certificate to their system but now there will at least be the option

**Comment by AndySchroder** [comment]
By `system`, do you mean their phone?

**Comment by kaloudis** [comment]
@AndySchroder yep, we'll create documentation that'll be linked to on the node config page and display a warning for users who chose not to use verification

**Comment by kaloudis** [comment]
target is v0.3.0 which should be the next release

**Comment by ZeusLN** [comment]
Fixed in v0.3.0
